### DISCUSSION TITLE
Why restart on no wifi?

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#18888] [Q&A]
As I read the code in support_wifi.ino line 868-869 
```
          if (100 == Wifi.max_retry) {  // Restart after 100 * (WIFI_RETRY_OFFSET_SEC + MAC) / 2 seconds
            TasmotaGlobal.restart_flag = 2;
```
It means that with the default wificonfig setting of 4, in the absence of wifi, Tasmota will dutifully restart once every 16.6-30 minutes (depending on last digit of MAC address). 

Is there a reason for this? I would prefer the unit to keep trying wifi forever, rather than racking up restarts and flash writes (85-ish a day).

I know Tasmota isn't built for no wifi, but is there a need for that restart?

**Comment by s-hadinger** [comment]
The reason it restarts is because loss of wifi could be that the wifi subsystem crashed (in esp-idf) and the only way to get it back is to restart. 

I agree that this is very rare, but if it happens to a remote device, you have no way to get it back online if it doesn't restart on its own. 

I'm not sure you can disable this behavior.
